### PR TITLE
build(nix): fix build issues

### PR DESCRIPTION
**Pull request by yusdacra**

### CODE REVIEW

**Review by pickfire — APPROVED** [review]
nixrdp gave a thumbs up so I guess it should be good

File: flake.nix [hunk 1]
```diff
@@ -28,27 +28,53 @@
...
+                  sha256 = "sha256-KayR7K7UC0mT6EjHsZsCYY9IVDJzft63fGpPKGSY8nQ=";
+                };
+              in
+              pkgs.runCommand prev.src.name { } ''
                 mkdir -p $out
                 ln -s ${prev.src}/* $out
                 ln -sf ${helix}/helix-syntax/languages $out
               '';
-            };
+            preConfigure = "mkdir -p ../runtime/grammars";
```
Comment (pickfire) [inline flake.nix:49]: Wait, is this necessary? Why do we need to create it here and override it with `ln -sf` later on?

Comment (yusdacra) [reply flake.nix:49]: If we don't do that `helix-syntax` won't build, as that's where it outputs the compiles grammars.

**Comment by archseer** [comment]
Thanks for the fix! I guess we need to do a similar change in nixpkgs? Is there any way I can make this simpler for nix packaging, maybe making the output directory configurable?

**Comment by yusdacra** [comment]
> Thanks for the fix! I guess we need to do a similar change in nixpkgs? Is there any way I can make this simpler for nix packaging, maybe making the grammar output directory configurable?

I think what could make this simpler is basically the ability to compile the grammars seperately. So the `build.rs` of helix-syntax would be made into a seperate crate. And the ability to turn off the compiling of grammars in helix-syntax to avoid compiling them twice. Although we probably don't need these for nixpkgs since we can make it output both a runtime dir with compiled grammars and the hx binary itself (with crate2nix I'm not sure if it is possible, since it would require propagating the helix-syntax output upwards to helix-term, hence why I compile helix-syntax twice). But still it would be nice to have those since I think people might be interested in only compiling for languages they want and so on.

**Pull request merged**